TESTING.MD

```diff
@@ -30,74 +24,159 @@ rake style:ruby:auto_correct  # Auto-correct RuboCop offenses
 
 Images include:
 
-* Windows Server 2008
-* Windows Server 2012
-* Windows 7 Professional
-* Windows 8.1 Professional
-* Mac OS X 10.7-10.11
-* Solaris 10.11
+- Windows Server 2008
+- Windows Server 2012
+- Windows 7 Professional
+- Windows 8.1 Professional
+- Mac OS X 10.7-10.11
+- Solaris 10.11
+- Integration Testing using Cloud providers
+
+Integration tests can be performed on cloud providers using Test Kitchen plugins. If the cookbook ships with a `.kitchen.cloud.yml` that references environmental variables present in the shell that `kitchen test` is ran from. These usually contain authentication tokens for driving IaaS APIs, as well as the paths to ssh private keys needed for Test Kitchen log into them after they've been created.
+
+Examples of environment variables being set in `~/.bash_profile`:
+
+```shell
+# digital_ocean
+export DIGITAL_OCEAN_CLIENT_ID='your_bits_here'
+export DIGITAL_OCEAN_API_KEY='your_bits_here'
+export DIGITAL_OCEAN_SSH_KEY_IDS='your_bits_here'
+
+# aws
+export AWS_ACCESS_KEY_ID='your_bits_here'
+export AWS_SECRET_ACCESS_KEY='your_bits_here'
+export AWS_KEYPAIR_NAME='your_bits_here'
+```
+
+Integration tests using cloud drivers can be performed with either
+
+```shell
+export KITCHEN_YAML=.kitchen.cloud.yml
+bundle exec kitchen test
+```
+
+or
+
+```shell
+rake integration:cloud
+```
+
+## Digital Ocean Hint
+
+At the time of this writing, you cannot find the numerical values needed for your SSH_KEY_IDS from the GUI. Instead, you will need to access the API from the command line.
+
+```shell
+curl -L 'https://api.digitalocean.com/ssh_keys/?client_id=your_bits_here&api_key=your_bits_here'
+```
+
+## Words about .travis.yml
+
+In order for Travis to perform integration tests on public cloud providers, two major things need to happen. First, the environment variables referenced by `.kitchen.cloud.yml` need to be made available. Second, the private half of the ssh keys needed to log into machines need to be dropped off on the machine.
+
+The first part is straight forward. The travis gem can encrypt environment variables against the public key on the Travis repository and add them to the .travis.yml.
+
+```
+        > Format markdown
+```
 
 If you are a Chef employee, you will need to get VMWare Fusion, [Vagrant plugin for VMWare](https://www.vagrantup.com/vmware/), and set up a [Hashicorp Atlas](https://www.hashicorp.com/atlas.html) account that has access to the Chef images.
 
-Once you have the required software, login to Atlas from the command line. Uncomment the suites in .kitchen.yml file, or set the environment variable KITCHEN_LOCAL_YAML to kitchen.vmware.yml.
+<<<<<<< a2522b87718a266938e656c7597c34e6fa1571f3 Once you have the required software, login to Atlas from the command line. Uncomment the suites in .kitchen.yml file, or set the environment variable KITCHEN_LOCAL_YAML to kitchen.vmware.yml.
```

Comment:
Is this "<<<<<<< a2522b87718a266938e656c7597c34e6fa1571f3 " actually in the text? That looks like some meta information accidentally getting into documentation.